pkgs/by-name/go/goto/package.nix

```diff
@@ -0,0 +1,65 @@
+{
+  coreutils,
+  fetchFromGitHub,
+  gawk,
+  goto,
+  lib,
+  nix-update-script,
+  runCommand,
+  stdenvNoCC,
+}:
+
+let
+  owner = "iridakos";
+  rev = "b7fda54e0817b9cb47e22a78bd00b4571011cf58";
+
+in
+stdenvNoCC.mkDerivation {
+  pname = "goto";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    inherit owner rev;
+    repo = "goto";
+    hash = "sha256-dUxim8LLb+J9cI7HySkmC2DIWbWAKSsH/cTVXmt8zRo=";
+  };
+
+  strictDeps = true;
+
+  buildInputs = [
+    coreutils
+    gawk
+  ];
+
+  postInstall = ''
+    install -Dm644 goto.sh -t $out/share/
+  '';
+
+  passthru.tests.basic-usage = runCommand "goto-basic-usage" { } ''
```

Comment:
Maybe this could be used as the `checkPhase`?

Reply:
Probably. But isn't the purpose of the `checkPhase` to run builtin tests?